### DISCUSSION TITLE
How to easily differentiate between "subdocument" & "nestedpath"?

### DISCUSSION THREAD

**Discussion by vanuverma** [Automattic/mongoose#11766] [Q&A]
I have a NestJs code where a schema is defined like below:   

`
import { Prop, raw, Schema } from "@nestjs/mongoose";
import { Document } from "mongoose";
import { ObjectId } from "mongodb";
import { CSharpDate, CSharpDateInterface } from "./utils";

@Schema()
export class UserConsent extends Document {
    @Prop({ required: true })
    UserAccountId!: string;

    @Prop()
    Result?: string;

    @Prop({ required: true })
    CorrelationId!: string;

    @Prop({ type: ObjectId, default: null })
    ProfileId?: ObjectId | null;

    @Prop(raw({ type: CSharpDate, required: true }))
    CreatedAt!: CSharpDateInterface;

    @Prop(raw({ type: CSharpDate, default: null }))
    UpdatedAt?: CSharpDateInterface | null;
}

`

and the CSharpDate type and CSharpDateInterface are defined as below (note that the CSharpDate does not have a @Schema() decorator on it):
`export interface CSharpDateInterface {
    DateTime: Date;
    Ticks: number;
    Offset: number;
}

export const CSharpDate = {
    DateTime: Date,
    Ticks: Number,
    Offset: Number,
};`

My question here: will the CreatedAt property be treated as "Subdocument" or "Nestedpath" by Mongoose? And why would it contain an "_id" field when saved into database collection?

**Comment by vkarpov15** [comment]
As a subdocument. You can disable `_id` by doing `@Prop(raw({ type: CSharpDate, required: true, _id: false }))` or `export const CSharpDate = { _id: false, ... }`
